### PR TITLE
[new release] ez-conf-lib, conf-gmp-paths, conf-mpfr-paths

### PR DESCRIPTION
**Pull request by nberth**
Temporary, improved versions of `conf-gmp` and `conf-mpfr`, renamed into `conf-gmp-paths` and `conf-mpfr-paths`.

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Queued on https://github.com/ocaml/opam-repository/pull/23613

**Comment by nberth** [comment]
The changes I suggested in #23613 may impact many packages and need many resources to be reviewed. So I'm now suggesting to consider this PR instead, to progress faster on related issues on several packages that depend on `conf-gmp` and `conf-mpfr`. Simply using new package names like `conf-gmp-paths`:
- have no consequences on packages that work already when using `conf-gmp` and `conf-mprf`;
- allow maintainers of said packages to manually opt-in and benefit from the variable names defined by the `-paths` versions.

**Comment by nberth** [comment]
The changes I suggested in #23613 may impact many packages and need many resources to be reviewed. So I'm now suggesting to consider this PR instead, to progress faster on related issues on several packages that depend on `conf-gmp` and `conf-mpfr`. Simply using new package names like `conf-gmp-paths`:
- have no consequences on packages that work already when using `conf-gmp` and `conf-mprf`;
- allow maintainers of said packages to manually opt-in and benefit from the variable names defined by the `-paths` versions.

**Comment by rmonat** [comment]
@mseri any update regarding these packages? I am interested in trying `conf-gmp-paths` and `conf-mpfr-paths` to make some academic project easily installable for MacOS users.

**Comment by mseri** [comment]
Sorry for the delay. Since my other attempts will surely take longer, they should not block this attempt: this PR is not modifying `conf-gmp` which is the delicate change that requires very careful testing, so let's go ahead.

File: packages/ez-conf-lib/ez-conf-lib.1/opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+     helps writing virtual packages used to detect system libraries"
+authors: ["Nicolas Berthier"]
+maintainer: "Nicolas Berthier <m@nberth.space>"
+homepage: "https://github.com/nberth/ez-conf-lib"
+dev-repo: "git+https://github.com/nberth/ez-conf-lib.git"
+bug-reports: "https://github.com/nberth/ez-conf-lib/issues"
+license: "GPL-3.0-only"
+build: [
+  [ "sh" "-ecx" "sed -e 's_@@ez-conf-lib:lib@@_%{_:lib}%_g' \
+                     ez-conf-lib.config.in > ez-conf-lib.config" ]
```
Comment (sim642) [inline packages/ez-conf-lib/ez-conf-lib.1/opam:14]: This is utterly broken for local opam switches in `_opam` directories: https://github.com/nberth/ez-conf-lib/issues/3.

Comment (nberth) [reply packages/ez-conf-lib/ez-conf-lib.1/opam:14]: Thanks for spotting that beginner mistake (arf… that's what happens when you hack things too quickly). #25313 should bring a version that does not rely on `sed`.

**Pull request merged**